libs/core/src/lib/message-box/message-box-header/message-box-header.component.ts

```diff
@@ -25,29 +29,43 @@ import {
 @Component({
     selector: 'fd-message-box-header',
     templateUrl: './message-box-header.component.html',
-    changeDetection: ChangeDetectionStrategy.OnPush,
-    providers: [{ provide: MESSAGE_BOX_CONFIGURABLE_ELEMENT, useExisting: MessageBoxHeaderComponent, multi: true }]
+    changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class MessageBoxHeaderComponent
-    extends DialogHeaderBase
-    implements AfterContentInit, MessageBoxConfigurableElement
-{
+export class MessageBoxHeaderComponent extends DialogHeaderBase implements AfterViewInit {
+    /** @hidden */
+    @ViewChild('defaultTemplateHeaderBar')
+    defaultTemplateHeaderBar?: ElementRef<Element>;
+
+    /** @hidden */
+    private _containsProjectedIcon = false;
+
+    /** @hidden */
+    get messageBoxConfig(): MessageBoxConfig {
+        return this.messageBox?._messageBoxConfig || {};
+    }
+
     /** @hidden */
-    constructor(@Optional() public messageBoxConfig: MessageBoxConfig, changeDetectorRef: ChangeDetectorRef) {
-        super(changeDetectorRef);
-        this.messageBoxConfig = this.messageBoxConfig || {};
+    constructor(private _cdr: ChangeDetectorRef, @Optional() private messageBox?: MessageBoxHost) {
+        super(_cdr);
     }
 
     /** @hidden */
     get _showSemanticIcon(): boolean {
         return (
-            (this.messageBoxConfig.type && this.messageBoxConfig.showSemanticIcon) ||
-            !!this.messageBoxConfig.customSemanticIcon
+            !this._containsProjectedIcon &&
+            ((this.messageBoxConfig.type && this.messageBoxConfig.showSemanticIcon) ||
+                !!this.messageBoxConfig.customSemanticIcon)
         );
     }
 
     /** @hidden */
-    ngAfterContentInit(): void {
-        super.ngAfterContentInit();
+    ngAfterViewInit(): void {
+        // running this in next microtask to avoid "ExpressionChangedAfterItHasBeenCheckedError"
+        Promise.resolve().then(() => {
+            this._containsProjectedIcon = !!this.defaultTemplateHeaderBar?.nativeElement.querySelector(
```

Comment:
Can we just use `@ContentChild(MessageBoxSemanticIcon)` & getter instead of doing calculations?

Reply:
you're completely right, thanks. done